lib/order_repository.rb

```diff
@@ -0,0 +1,45 @@
+require_relative 'order'
+
+class OrderRepository
+  def all
+    sql = 'SELECT * FROM orders'
+    result = DatabaseConnection.exec_params(sql, [])
+    
+    orders = []
+    result.each do |record|
+      order = Order.new
+      order.id = record['id'].to_i
+      order.customer = record['customer']
+      order.date = record['date']
+      orders << order
+    end
+    return orders
+  end
+
+  def create(order)
+    sql = 'INSERT INTO orders (customer, date) VALUES ($1, $2)'
+    params = [order.customer, order.date]
+    result = DatabaseConnection.exec_params(sql, params)
+    return nil
+  end
+
+  def find_with_items(id)
+    sql = 'SELECT orders.customer, orders.date, items.name
```

Comment:
Formatting of the sql query just for looks could be easier to read, but good working code!